### ISSUE TITLE
Unique field ignored when creating User via REST

### ISSUE THREAD

**Issue by notchris** [payloadcms/payload#873]
### Discussed in https://github.com/payloadcms/payload/discussions/872

<div type='discussions-op-text'>

<sup>Originally posted by **notchris** August  2, 2022</sup>
Hello! Thanks again for maintaining payloadcms @jmikrut 

When creating a new user via a REST API call, I tested creating a user with a duplicate 'username'. I've marked this field as unique in the field setup, but it allowed me to create the additional user. Am I setting the field unique property incorrectly?

Thank you!

Chris
![Screen Shot 2022-08-02 at 1 26 33 PM](https://user-images.githubusercontent.com/1944433/182437233-2a293629-48ff-405c-94c6-709e40136885.png)

```ts
const Users: CollectionConfig = {
  slug: 'users',
  auth: true,
  admin: {
    useAsTitle: 'username',
  },
  hooks: {
    afterLogin: [userLoginHook],
  },
  access: {
    read: adminOrSelf,
    update: adminOnly,
    create: () => true,
  },
  fields: [
    {
      name: 'username',
      label: 'Username',
      type: 'text',
      unique: true,
      required: true,
      minLength: 5,
      maxLength: 16
    },
    {
      name: 'banned',
      label: 'Ban User',
      type: 'checkbox',
      admin: {
        position: 'sidebar',
      },
    },
  ],
}
```

Example of fetch request:

```js
				const res = await fetch('http://localhost:3000/api/users/', {
					method: 'POST',
					headers: {
						'Content-Type': 'application/x-www-form-urlencoded'
					},
					credentials: 'include',
					body: new URLSearchParams({
						username: v.username,
						email: v.email,
						password: v.password
					})
				})
				```

 </div>

**Comment by DanRibbens** [comment]
Hi @notchris,

Your username field is correct, we had a bug that was fixed in the current release v1.0.12.

Let us know if you if you have any trouble.

**Comment by stnokott** [comment]
Might need reopening due to reports in discussion #872
